src/reflection.jl

```diff
@@ -0,0 +1,34 @@
+import InteractiveUtils
+export @ka_code_typed
+
+function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing)
+    # get the iterspace and dynamic of a kernel
+    ndrange, workgroupsize, iterspace, dynamic = KernelAbstractions.launch_config(kernel, ndrange, workgroupsize)
+    # get the first block
+    block = @inbounds KernelAbstractions.blocks(iterspace)[1]
+    # get a context of the kernel based on the first block
+    ctx = KernelAbstractions.mkcontext(kernel, block, ndrange, iterspace, dynamic)
+    # reformat
+    if argtypes isa Type
+        argtypes = argtypes.parameters
+    end
+    # use code_typed
+    return InteractiveUtils.code_typed(KernelAbstractions.Cassette.overdub, (typeof(ctx), typeof(kernel.f), argtypes...))
```

Comment:
```suggestion
    return InteractiveUtils.code_typed(KernelAbstractions.Cassette.overdub, (typeof(ctx), typeof(kernel.f), argtypes...); kwargs...)
```

`code_typed` takes a couple of key-word arguments, that control the output.

Reply:
> Excellent! This is looking good. Can you add a test?
> 
> Ideally we also would have an interactive mode like, https://github.com/JuliaGPU/GPUCompiler.jl/blob/177bff39877f39c925e340cef391c1e11bfa28c4/src/reflection.jl#L48-L56

I can look into adding the interactive mode. I wasn't sure if that was the goal for this or if was only to have a functioning method.
I can also add a test, but I'm not sure what I should check. Just that it functions for a few different kind of calls, or that it produces a specific output? If you could link me to the @code_typed tests, I could try to replicate those to some degree.

Reply:
Yeah something simple like https://github.com/JuliaLang/julia/blob/b884dcad9598f862ab6c08c01e2800942769470c/stdlib/InteractiveUtils/test/runtests.jl#L127

But where you test the different KA keyword arguments. 

> I can look into adding the interactive mode. I wasn't sure if that was the goal for this or if was only to have a functioning method.

Interactive mode is definitely a stretch goal.